recipe/recipe.yaml

```diff
@@ -24,6 +24,9 @@ requirements:
   run:
     - python >=${{ python_min }}
     - numba >=0.59.1
+    - cuda-version >=11.2
+    - cuda-bindings  # Includes nvrtc, nvvm, libdevice
+    - cuda-cudart-dev  # Includes cudadevrt
```

Comment:
@gmarkall I know that we have all of the shared libraries covered based on these but I'm not clear on what headers we'd look for at runtime. Given the current state doesn't enforce any specific dependencies, we aren't making things worse, but it would be nice to capture whatever headers we need as dependencies as well.

Reply:
For "normal usage" we don't have extra requirements:

* The headers we need are vendored (cuda_fp16.h / hpp, cuda_bf16.h / hpp).
* We require `cuda/atomic` for when NRT is enabled (it is experimental, and disabled by default).
* Others we require / expect only for the test suite are:
  - `cooperative_groups.h`
  - `cuda_device_runtime_api.h`
  - `curand_kernel.h`